### PR TITLE
[FLINK-31309] Delete DFS schema when create table failed in HiveCatalog

### PR DESCRIPTION
**Pull request by FangYongs**
Delete DFS schema when create table failed in HiveCatalog

### CODE REVIEW

File: flink-table-store-hive/flink-table-store-hive-catalog/src/test/java/org/apache/flink/table/store/hive/HiveCatalogITCase.java [hunk 1]
```diff
@@ -527,6 +530,38 @@ public void testQuickPathInShowTables() throws Exception {
...
+        tEnv.executeSql(
+                String.join(
+                        "\n",
+                        "CREATE CATALOG my_hive_custom_client WITH (",
+                        "  'type' = 'table-store',",
+                        "  'metastore' = 'hive',",
+                        "  'uri' = '',",
+                        "  'warehouse' = '" + path + "',",
+                        "  'metastore.client.class' = '"
+                                + TestHiveMetaStoreClient.class.getName()
```
Comment (JingsongLi) [inline flink-table-store-hive/flink-table-store-hive-catalog/src/test/java/org/apache/flink/table/store/hive/HiveCatalogITCase.java:544]: Can you create a new class `FailMetaStoreClient`?

File: flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java [hunk 2]
```diff
@@ -241,6 +244,12 @@ public void createTable(Identifier identifier, Schema schema, boolean ignoreIfEx
         try {
             client.createTable(table);
         } catch (TException e) {
+            Path path = getDataTableLocation(identifier);
+            try {
+                fileIO.delete(path, true);
```
Comment (JingsongLi) [inline flink-table-store-hive/flink-table-store-hive-catalog/src/main/java/org/apache/flink/table/store/hive/HiveCatalog.java:249]: Maybe introduce a method in `FileIO`, `deleteDirectoryQuietly`?

**Review by JingsongLi — COMMENTED** [review]
Thanks @FangYongs , left two comments.

**Comment by FangYongs** [comment]
Thanks @JingsongLi Done

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**